lib/io.c

```diff
@@ -11,6 +11,7 @@
 
 void metal_io_init(struct metal_io_region *io, void *virt,
 	      const metal_phys_addr_t *physmap, size_t size,
+	      const metal_phys_addr_t *physmap_drv,
```

Comment:
This will break the existing API in the next release. When possible, we try to use a deprecation mechanism.

Here we also have the increase of th function parameters, which seems already  quite significant to me. So perhaps it is a good time to rework this API.

I would create a `metal_io_config `structure that would contain these parameters. Then, create a new function `void metal_io_init_region(struct metal_io_region *io, const struct metal_io_config *cfg)`. Finally, I would update `metal_io_init` to use `metal_io_init_region` and tag `metal_io_init_region` as deprecated.

But perhaps someone else as an alternative to propose?

Reply:
This looks like good idea. Advantage is we can extend structure as needed but keep API interface the same.

Reply:
@arnopo @tnmysh I've decided to add 2 new members: `metal_io_ops::remote_phys_to_offset` and `metal_io_ops::offset_to_remote_phys`.

The changes in libmetal will be similar to this commit: https://github.com/OpenAMP/libmetal/commit/a90bbd8fb6a8045947dd8ac8a7e48691cfbc3298

For context see also [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/84170#issuecomment-2639827199).

Reply:
Do you need to create 2 new ops?  Or could you reuse phys_to_offset and offset_to_phys?

Look to me that based on the io address you could decide to apply or not the memory translation

Reply:
Yes, you're right @arnopo. I can use those.
Therefore, I think this PR can be closed.

[This](https://github.com/OpenAMP/libmetal/pull/322#discussion_r1942489096) can be added to the backlog, as an improvement, for later.